src/map/storage.c

```diff
@@ -212,21 +220,29 @@ int storage_additem(struct map_session_data* sd, struct item* item_data, int amo
  *------------------------------------------*/
 int storage_delitem(struct map_session_data* sd, int n, int amount)
 {
+	struct item *it = NULL;
+
 	nullpo_retr(1, sd);
-	Assert_retr(1, n >= 0 && n < MAX_STORAGE);
-	if( sd->status.storage.items[n].nameid == 0 || sd->status.storage.items[n].amount < amount )
+
+	Assert_retr(1, n >= 0 && n < VECTOR_LENGTH(sd->storage));
+
+	if (sd->state.storage_flag != STORAGE_FLAG_NORMAL)
```

Comment:
Can you explain the reason for this line? Do we want delitem to silently fail (and return success) if the storage isn't open? I believe we have some functions that don't change the sd's storage_flag, but still want to delete items from the storage.